lib/plugin/list.js

```diff
@@ -0,0 +1,265 @@
+var isUnorderedItem = function (line) {
+  var match = line.match(/^( *)[-|\*|\+] /)
+  return match
+}
+
+var isOrderedItem = function (line) {
+  var match = line.match(/^( *)([1-9][0-9]*)\. /)
+  return match
+}
+
+var prefixCodefence = function (line) {
+  if (line.length < 3) return 0
+
+    // This is how it looks in redcarpet and I didn't feel like making it better
```

Comment:
The weirdest part is that they use ```while (i < 4 && line[i] === ' ') i++``` just 40 lines later. But hey, it beats the bit-flags they pass through every parsing function.